src/routes/Measurements/components/nettests/Dash.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react'
+import { FormattedMessage } from 'react-intl'
+
+import { mlabServerToCountry, mlabServerToName } from '../../../../util/nettest'
+
+/*
+ * This table is derived from:
+ * https://support.google.com/youtube/answer/1722171?hl=en-GB
+ */
+
+const minimumBitrateForVideo = [
+  {
+    'sfr_min_bitrate': 600,
+    'hfr_min_bitrate': 1000,
+    'type': '240p'
+  },
+  {
+    'sfr_min_bitrate': 1000,
+    'hfr_min_bitrate': 1500,
+    'type': '360p'
+  },
+  {
+    'sfr_min_bitrate': 2500,
+    'hfr_min_bitrate': 4000,
+    'type': '480p'
+  },
+  {
+    'sfr_min_bitrate': 5000,
+    'hfr_min_bitrate': 7500,
+    'type': '720p (HD)'
+  },
+  {
+    'sfr_min_bitrate': 8000,
+    'hfr_min_bitrate': 12000,
+    'type': '1080p (full HD)'
+  },
+  {
+    'sfr_min_bitrate': 16000,
+    'hfr_min_bitrate': 24000,
+    'type': '1440p (2k)'
+  },
+  {
+    'sfr_min_bitrate': 35000,
+    'hfr_min_bitrate': 53000,
+    'type': '2160p (4k)'
+  },
+]
+
+const getOptimalVideoRate = (test_keys) => {
+  let optimalRate = minimumBitrateForVideo[0]
+
+  minimumBitrateForVideo.forEach((rate) => {
+    if (test_keys.simple.median_bitrate >= rate['sfr_min_bitrate']) {
+      optimalRate = rate
+    }
```

Comment:
Ok sorry, perhaps it was not clear what this function was supposed to do.

It doesn't actually check to see if it fits within the boundary of `hfr_min_bitrate` and `sfr_min_bitrate` (from your comment I determine you were assuming this is what it does). In fact it doesn't even use the High Frequency Bitrate value in the computation. I just took note of it as I may be interested in using it in the future, but maybe to avoid confusion it should just be dropped.

The point of the function is to figure out what is the maximum supported video quality based on the bitrate returned by the dash test. So it actually is not under-reporting and there is no possible way to have there be an overlap.

Upon further inspection of how your refactored function works, it's basically equivalent to just returning `240p` every time.

No matter what value it's given in input it will always return `240p`, to expand you above example:

```
const minimumBitrateForVideo = [
  {
    'sfr_min_bitrate': 600,
    'hfr_min_bitrate': 1000,
    'type': '240p'
  },
  {
    'sfr_min_bitrate': 900,  // Note: modified (originally 1000) to explain my point
    'hfr_min_bitrate': 1500,
    'type': '360p'
  },
  {
    'sfr_min_bitrate': 35000,
    'hfr_min_bitrate': null, // not relevant
    'type': '2160p (4k)'
  },

  // [...] removed other entries because not relevant to my point
]

const getOptimalVideoRate = (test_keys) => {
  let optimalRate = minimumBitrateForVideo[0]

  minimumBitrateForVideo.forEach((rate) => {
    if (test_keys.simple.median_bitrate >= rate['sfr_min_bitrate']) {
      optimalRate = rate
    }
  })
  return optimalRate
}

const getOptimalVideoRateEx = (test_keys) => {
  let optimalRate = null;
  minimumBitrateForVideo.forEach((rate) => {
    // Make sure we select the lowest speed bucket in case they overlap
    if (optimalRate === null &&
        test_keys.simple.median_bitrate >= rate['sfr_min_bitrate']) {
      optimalRate = rate
    }
  })
  if (optimalRate === null) {
    optimalRate = minimumBitrateForVideo[0]
  }
   return optimalRate
}

const input = {
  "simple": {
    "median_bitrate": 911,
  },
}
const inputTwo = {
  "simple": {
    "median_bitrate": 2000000000, // MUCH bandwidth
  },
}

console.log("orig")
console.log(getOptimalVideoRate(input).type)
console.log(getOptimalVideoRate(inputTwo).type)
console.log("ex")
console.log(getOptimalVideoRateEx(input).type)
console.log(getOptimalVideoRateEx(inputTwo).type)
```

results in:

```
orig
360p
2160p (4k)
ex
240p
240p
```